client/lib/plans/personal-plan.js

```diff
@@ -54,13 +81,18 @@ export const insertSitePersonalPlan = plans => {
 		has_domain_credit,
 		product_name,
 		product_slug,
-		raw_price
-	} = personalPlan;
+		raw_price,
+		currency_code
+	} = formatPlan( plans );
 
 	if ( ! ( isEmpty( plans ) || has( plans, product_id ) ) ) {
+		const hasCurrentPlan = some( plans, matchesProperty( 'current_plan', true ) );
```

Comment:
Can we not use the existing selector?

`getCurrentPlan = ( state, siteId )`

https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/plans/selectors.js#L34